sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py

```diff
@@ -183,10 +183,17 @@ def add_node(self, node: Union[BaseNode, AutoMLJob]):
         """
         self.nodes.append(node)
 
-    def build(self, non_pipeline_params_dict=None) -> PipelineComponent:
+    def build(self, original_kwargs, non_pipeline_params_dict=None) -> PipelineComponent:
```

Comment:
"original_kwargs" is not a clear name to me, maybe "user_provided_kwargs" ? Also, add a '*' into the parameters.